clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -107,4 +129,23 @@ public ByteBuffer payload() {
         return this.buffer;
     }
 
+    // We determine if a peer connection uses SSL/TLS by seeing from the first few bytes.
+    private void ensureNotHandshakeFailurePacket(ReadableByteChannel channel, ByteBuffer size) throws IOException {
+        if (ensureThrough)
+            return;
+        ensureThrough = true;
+        String head = Utils.hexToString(size.array());
+        if (SSL_HANDSHAKE_ALERT.matcher(head).find()) {
```

Comment:
This feels like an expensive way to check a byte sequence. Wouldn't it be cheaper to compare byte arrays?